packages/logs/src/domain/logger.ts

```diff
@@ -39,29 +51,47 @@ export class Logger {
   }
 
   @monitored
-  log(message: string, messageContext?: object, status: StatusType = StatusType.info) {
-    this.handleLogStrategy({ message, context: deepClone(messageContext) as Context, status }, this)
+  log(message: string, messageContext?: object, status: StatusType = StatusType.info, error?: Error) {
+    let context: Context
+
+    if (status === StatusType.error || (error !== undefined && error !== null)) {
+      // Always add origin if status is error (backward compatibility), or we have an actual error
+      const errorContext = { error: { origin: ErrorSource.LOGGER } as Context }
+
+      // Extract information from error object if provided
+      if (error instanceof Error) {
+        const stackTrace = computeStackTrace(error)
+        errorContext.error.kind = stackTrace.name
+        errorContext.error.message = stackTrace.message
+        errorContext.error.stack = toStackTraceString(stackTrace)
+        // Serialize other types if provided as error parameter
+      } else if (error !== undefined && error !== null) {
+        errorContext.error.message = `${PROVIDED_ERROR_MESSAGE_PREFIX} ${JSON.stringify(error, undefined, 2)}`
+        errorContext.error.stack = NO_ERROR_STACK_PRESENT_MESSAGE
+      }
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏it could be nice to use `computeRawError` here so we have the same logic for serializing errors between RUM and Logs (ex: same empty message fallback, same non-`Error` instance serialization, `error.causes` support...)
* Right now, `computeRawError` requires a `StackTrace` as parameter, but I believe this could be changed and the `StackTrace` could be computed directly within `computeRawError` instead.
* It also requires unneeded things in this context, like `startClocks`. This might be adjusted, maybe with the new intermediary function that requires less parameters.
* The mapping between `RawError` to the logs error context could be done here to handle subtelties like `type -> kind`.

Reply:
Agreed! Will be done in a separate PR as this needs a full review of the computeRawError test cases.

Reply:
why not doing the exact same thing as in DD_RUM.addError (with logs specific): 
https://github.com/DataDog/browser-sdk/blob/1cc4fc73063d43f3abad1194bf069a992a8003d5/packages/rum-core/src/domain/rumEventsCollection/error/errorCollection.ts#L69-L78
and then build the `errorContext` from the needed `rawError` properties, something like:
```ts
const errorContext = {
  origin: ErrorSource.LOGGER,
  message: rawError.message,
  kind: rawError.type,
  stack: rawError.stack
}
```